packages/angular/src/utils/mf/mf-webpack.ts

```diff
@@ -65,17 +65,37 @@ export function shareWorkspaceLibraries(
       continue;
     }
 
-    collectWorkspaceLibrarySecondaryEntryPoints(
-      key,
-      join(workspaceRoot, library.root),
-      tsconfigPathAliases
-    ).forEach(({ name, path }) =>
-      pathMappings.push({
-        name,
-        path,
-      })
+    const needsSecondaryEntryPointsCollected = existsSync(
+      join(workspaceRoot, library.root, 'ng-package.json')
     );
 
+    if (needsSecondaryEntryPointsCollected) {
+      const tsConfigAliasesForLibWithSecondaryEntryPoints = Object.entries(
+        tsconfigPathAliases
+      ).reduce((acc, [tsKey, tsPaths]) => {
+        if (!tsKey.startsWith(library.importKey)) {
+          return { ...acc };
+        }
+
+        if (tsPaths.some((path) => path.startsWith(`${library.root}/`))) {
+          acc = { ...acc, [tsKey]: tsPaths };
+        }
+
+        return acc;
+      }, {});
```

Comment:
There's already some sort of filtering inside `collectWorkspaceLibrarySecondaryEntryPoints` which becomes useless with this change. I think it would be better to move this filter inside that function and replace the old filter that's there. Wdyt?

Reply:
Yeah I think that makes sense, I updated it